ci/fix-renovate-args.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+if [ -z "$1" ]; then
+  echo "Error: Path to Dockerfile is required as the first argument."
+  exit 1
+fi
+DOCKERFILE_PATH="$1"
+
+CUDA_VER="${CUDA_VER:-11.2.0}"
+LINUX_VER="${LINUX_VER:-ubuntu20.04}"
+PYTHON_VER="${PYTHON_VER:-3.8}"
+ARCH="${ARCH:-x86_64}"
+MANYLINUX_VER="${MANYLINUX_VER:-manylinux_2_28}"
+
+YAML_FILE="renovate.yaml"
+
+BUILD_ARGS=""
+
+if [ -f "$YAML_FILE" ]; then
+  while IFS= read -r line; do
+    key=$(echo "$line" | cut -f1 -d':')
+    value=$(echo "$line" | cut -f2 -d':')
+    if [ -n "$GITHUB_ACTIONS" ]; then
+      # Format the output for array appending (in ci/compute-build-args.sh)
+      BUILD_ARGS+="$key=$value "
+    else
+      # Locally, format as Docker build arguments
+      BUILD_ARGS+="--build-arg $key=$value "
+    fi
+  done < <(yq e '. | to_entries | .[] | .key + ":" + (.value | sub("^v"; ""))' "$YAML_FILE")
+fi
+
+if [ -n "$GITHUB_ACTIONS" ]; then
+  echo "$BUILD_ARGS"
+else
+  eval "docker build -f \"$DOCKERFILE_PATH\" . \
+    --build-arg CUDA_VER=\"$CUDA_VER\" \
+    --build-arg LINUX_VER=\"$LINUX_VER\" \
+    --build-arg PYTHON_VER=\"$PYTHON_VER\" \
+    --build-arg CPU_ARCH=\"$ARCH\" \
+    --build-arg REAL_ARCH=\"$(arch)\" \
+    --build-arg MANYLINUX_VER=\"$MANYLINUX_VER\" \
+    $BUILD_ARGS"
+fi
```

Comment:
I think this file might need some more thought...

It looks like in GHAs, it will echo build arguments but outside of CI it will perform a `docker build`.

I'm not sure we want a single file that performs two totally different functions depending on whether it's run in GHAs or not.

Reply:
Yes. It's the implementation that I think needs to be reconsidered.

As a contributor to the repository, it's not intuitive to me that running a script called `fix-renovate-args.sh` will perform a Docker build.

Reply:
I think a better implementation could be a Makefile with a `build` target.

The `build` target will run `docker build` and call a script under the hood to parse `versions.yaml` and inject the versions as build arguments. The target can also pass any additional positional parameters to the `docker build`  command.

An example might look like...

```sh
make build -f Dockerfile context/
```

Reply:
alternatively, a copy-pasteable README snippet is probably also sufficient. This would avoid the need to create a wrapper script.

e.g:

```sh
docker build $(ci/compute-build-args.sh) -f Dockerfile context/
```